tests/end-to-end/_files/with_environment_variable/SomeTest.php

```
@@ -0,0 +1,124 @@
+        $this->testFOOShouldHaveValueFromClassAttribute1();
+    }
+
+    #[WithEnvironmentVariable('BAR', 'bar')]
+    public function testBARShouldHaveValueFromMethodAttribute(): void
+    {
+        $this->assertSame('bar', $_ENV['BAR']);
+        $this->assertSame('bar', getenv('BAR'));
+    }
+
+    #[Depends('testBARShouldHaveValueFromMethodAttribute')]
+    public function testBARShouldHaveBeenRestoredToDefaultValue1(): void
+    {
+        $this->assertSame('2', $_ENV['BAR']);
+        $this->assertSame('2', getenv('BAR'));
+    }
+
+    #[WithEnvironmentVariable('BAR')]
+    public function testBARShouldHaveValueErasedFromMethodAttribute(): void
+    {
+        $this->assertFalse(isset($_ENV['BAR']));
+        $this->assertFalse(getenv('BAR'));
+    }
+
+    #[Depends('testBARShouldHaveValueErasedFromMethodAttribute')]
+    public function testBARShouldHaveBeenRestoredToDefaultValue2(): void
+    {
+        $this->testBARShouldHaveBeenRestoredToDefaultValue1();
+    }
+
+    #[WithEnvironmentVariable('BAZ', 'baz')]
+    public function testBAZShouldHaveValueFromMethodAttribute(): void
+    {
+        $this->assertSame('baz', $_ENV['BAZ']);
+        $this->assertSame('baz', getenv('BAZ'));
+    }
+
+    #[Depends('testBAZShouldHaveValueFromMethodAttribute')]
+    public function testBAZShouldHaveBeenErased(): void
+    {
+        $this->assertFalse(isset($_ENV['BAZ']));
+        $this->assertFalse(getenv('BAZ'));
+    }
+
+    #[RunInSeparateProcess]
+    #[WithEnvironmentVariable('BAR', 'bar')]
+    public function testRunInSeparateProcess(): void
+    {
+        $this->assertSame('foo', $_ENV['FOO']);
+        $this->assertSame('foo', getenv('FOO'));
+
+        $this->assertSame('bar', $_ENV['BAR']);
+        $this->assertSame('bar', getenv('BAR'));
+    }
+
+    #[WithEnvironmentVariable('BAZ', '1')]
+    #[WithEnvironmentVariable('BAZ', '2')]
+    #[WithEnvironmentVariable('BAZ', '3')]
+    public function testMultipleAttributesKeepTheLastValue(): void
```

maybe we should throw an exception for this case?